server/test/mocks/mock_data_store.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef INS_SERVER_TEST_MOCKS_DATA_STORE_HPP
+#define INS_SERVER_TEST_MOCKS_DATA_STORE_HPP
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "data_store.hpp"
+
+namespace ins_service
+{
+
+class MockDataStore;
+
+extern ::testing::NiceMock<MockDataStore>* g_mocked_data_store_;
```

Comment:
Can you explain as to why do we need this? Same for the other similar occurrences.

Reply:
Oh just realized it was a NiceMock... Nvm the comment! :D